File: algorithms/count_negative_numbers_in_a_sorted_matrix.rs
impl Solution {
    pub fn count_negatives(grid: Vec<Vec<i32>>) -> i32 {
        let mut ans = 0;
        let mut end = grid[0].len();
        for i in 0..grid.len() {
            let mut j = 0;
            while j < end && grid[i][j] >= 0 {
                j += 1;
            }
            ans += grid[i].len() - j;
            end = j;
        }
        ans as i32
    }
}

File: algorithms/easy/count_negative_numbers_in_a_sorted_matrix.kt
class Solution {
  fun countNegatives(grid: Array<IntArray>): Int =
      grid.sumOf {
        var l = 0
        var r = it.size - 1
        while (l <= r) {
          var m = l + (r - l) / 2
          if (it[m] >= 0) {
            l = m + 1
          } else {
            r = m - 1
          }
        }
        it.size - l
      }
}
